test/tests/functional/pbs_multi_sched.py

```diff
@@ -647,6 +647,268 @@ def test_multi_fairshare(self):
         self.assertEquals(n.nshares, sc3_shares)
         self.assertEquals(n.usage, sc3_usage)
 
+    @timeout(300)
+    def test_fairshare_usage(self):
+        """
+        Test the schedulers fairshare usage file and
+        check the usage file is updating correctly or not
+        """
+        self.common_setup()
+        rv = self.server.add_resource('foo', 'float', 'nh')
+        self.assertTrue(rv)
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'}, expect=True)
+        # Add entry to the resource group of default scheduler
+        self.scheds['default'].add_to_resource_group('grp1', 100, 'root', 60)
+        self.scheds['default'].add_to_resource_group('grp2', 200, 'root', 40)
+        self.scheds['default'].add_to_resource_group('pbsuser1',
+                                                     101, 'grp1', 40)
+        self.scheds['default'].add_to_resource_group('pbsuser2',
+                                                     102, 'grp1', 20)
+        self.scheds['default'].add_to_resource_group('pbsuser3',
+                                                     201, 'grp2', 30)
+        self.scheds['default'].add_to_resource_group('pbsuser4',
+                                                     202, 'grp2', 10)
```

Comment:
You do not need to test this for default scheduler. I believe that is already being tested as part of smoke test.

Reply:
Yes. No need to test default scheduler's fairshare here.